package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py

```diff
@@ -720,6 +741,8 @@ def _single_frame(self):
 
```

Comment:
I find it somewhat inelegant that `_get_dh_pairs`, a frame-wise operation, performs atom selection to create new donors and hydrogens objects. Ideally, given that `updating=self.update_selections` applies to both hydrogens and donors, this should not be necessary.